pkg/apis/configuration/v1alpha1/types.go

```diff
@@ -31,6 +31,7 @@ type Upstream struct {
 	LBMethod            string      `json:"lb-method"`
 	FailTimeout         string      `json:"fail-timeout"`
 	MaxFails            *int        `json:"max-fails"`
+	MaxConns            *int        `json:"max-conns"`
```

Comment:
Why do we need a pointer here? 🤔

Reply:
@Rulox My thinking was if you set MaxConns globally via configmap you may want to overwrite for one vs/vsr back to `0` (equivalent to unset).

Reply:
Yup ok.